app/app.js

```diff
@@ -137,31 +137,33 @@ window.run = function() {
             });
         };
 
-        var appLayoutPromise = Application.getOSInformation().then(function(osInfo) {
-            if (BaseConfig.useTabLayout) {
-                return createTabBarLayout(counterBadgeControllerPromise);
-            } else {
-                return createDrawerLayout(counterBadgeControllerPromise);
-            }
-        });
+        var createLayout = function() {
+            return BaseConfig.useTabLayout
+                ? createTabBarLayout()
+                : createDrawerLayout();
+        };
 
-        // Show welcome modal only after layout is created for proper
-        // bookkeeping of the active view.
-        Promise.join(
-            appLayoutPromise,
-            welcomeModalControllerPromise,
-        function(menuController, welcomeModalController) {
-            // The welcome modal can be configured to show only
-            // once -- on initial startup, by passing in the
-            // parameter `{forced: false}` below
-            welcomeModalController.show({forced: true});
-
-            // Deep linking services will enable deep linking on startup
-            // and while running
-            // It will open the deep link in the current active tab
-            deepLinkingServices = new DeepLinkingServices(menuController);
-        });
+        var initMainLayout = function() {
+            return createLayout().then(function(menuController) {
```

Comment:
on a similar note, if you do make a change here, can we also change `menuConfig.js` to be in line with `drawerController.js` I feel like those should both be `menu` or both be `drawer`. or maybe the config should just be named `drawerMenuConfig.js` since the `drawerController` really controls more than just the left menu? Thoughts?